third_party/blink/renderer/platform/loader/fetch/resource.cc

```diff
@@ -656,6 +661,8 @@ void Resource::AddFinishObserver(ResourceFinishObserver* client,
 
   WillAddClientOrObserver();
   finish_observers_.insert(client);
+  if (recordreplay::IsRecordingOrReplaying("avoid-weak-pointers"))
+    replay_finish_observers_strong_.insert(client);
```

Comment:
Good point. I tried to time myself, and ended up rushing things too much. Will not sacrifice accuracy for time in the future.

Reply:
Please don't ever rush these patches.  Problems with these patches will be difficult to track down, and you might not be the one who has to do that.

Reply:
I went back and swept all possible modifications of all 4 datastructures. I overlooked 2 things:

1. `TriggerNotificationForFinishObservers` schedules `NotifyFinishObservers` using `WeakMember`. I added strong references to keep them alive until after `NotifyFinishObservers` exits.
2. In `RemoveFinishObserver` I overlooked `finish_observers_.erase`

I also double-checked the more complicated `FinishPendingClients`:
* The call to `clients_.insert` does not need an extra modification of `replay_clients_strong_`, since any such member would have had to come in via `clients_awaiting_callback_.insert(client)` in `AddClient`, which is already covered.